### PR TITLE
For #11903: Enforce private theme in custom tabs

### PR DESCRIPTION
**Pull request by NotWoods**
Attempt #2. Reverts the new drawable file from my previous PR and makes sure the theme is set properly instead.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by AaronMT** [comment]
I'm checking with Firebase Test-Lab team about the inconclusive result here.

File: app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt [hunk 1]
```diff
@@ -872,6 +875,14 @@ abstract class BaseBrowserFragment : Fragment(), UserInteractionHandler, Session
         }
     }
 
+    /**
+     * Set the activity normal/private theme to match the current session.
+     */
+    private fun updateThemeForSession(session: Session) {
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt:881]: We can probably remove the fun in HomeActivity then if you moved it here? I think these are the only two calls to it.

**Comment by liuche** [comment]
Failing unit tests: https://firefoxci.taskcluster-artifacts.net/SKbRep-OSfCnuuIH7KG6GQ/0/public/reports/test/testGeckoNightlyDebugUnitTest/index.html

and a couple of failing UI tests (not sure if they seem like they have anything to do with it):
saveLoginFromPromptTest: https://console.firebase.google.com/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/8139671025552175865/executions/bs.ab9c6dcfcd93f4c6/testcases/2/test-cases
goBackTest: https://console.firebase.google.com/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/8139671025552175865/executions/bs.5719b545a06ab5db/test-cases

**Comment by ekager** [comment]
Couldn't repro the unit test failures locally so trying a rebase 🤔

**Pull request merged**